From d6268b54ea4eb1433198a5d6b93191b26e9d01ec Mon Sep 17 00:00:00 2001
From: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Date: Tue, 20 Aug 2024 13:17:05 +0300
Subject: [PATCH] dts: Enable eMMC HS400 mode in case of verifiedboot and
 quickboot
Content-Type: text/plain; charset=UTF-8

Since verifiedboot support is based on Linux Kernel dtb its files,
the runtime fdt fixup mechanism from U-Boot will not work.

Hence, we statically remove the 'no-1-8-v' property which disables
eMMC HS400 mode support, in case of 'verifiedboot' and 'quick-boot'
features are both enabled.

Issue: ALB-11494
Signed-off-by: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
---
 arch/arm64/boot/dts/freescale/s32g274a-rdb2.dtsi | 9 ---------
 arch/arm64/boot/dts/freescale/s32g399a-rdb3.dtsi | 4 ----
 arch/arm64/boot/dts/freescale/s32gxxxa-evb.dtsi  | 1 -
 3 files changed, 14 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/s32g274a-rdb2.dtsi b/arch/arm64/boot/dts/freescale/s32g274a-rdb2.dtsi
index 30971db22f81..cf85fbc0d040 100644
--- a/arch/arm64/boot/dts/freescale/s32g274a-rdb2.dtsi
+++ b/arch/arm64/boot/dts/freescale/s32g274a-rdb2.dtsi
@@ -45,13 +45,4 @@ &pfe_netif1 {
 
 &usdhc0 {
 	disable-wp;
-	/* Remove no-1-8-v to enable higher speed modes for SD card.
-	 * However, this is not enough to enable HS400 or HS200 modes for eMMC.
-	 * In this case, the position of the resistor R797 must be changed
-	 * from A to B before removing the property.
-	 * If the property is removed without changing the resistor position,
-	 * HS*00 may be enabled, but the interface might be unstable because of
-	 * the wrong VCCQ voltage applied to the eMMC.
-	 */
-	no-1-8-v;
 };
diff --git a/arch/arm64/boot/dts/freescale/s32g399a-rdb3.dtsi b/arch/arm64/boot/dts/freescale/s32g399a-rdb3.dtsi
index 53f5a6752c24..ac60929947f7 100644
--- a/arch/arm64/boot/dts/freescale/s32g399a-rdb3.dtsi
+++ b/arch/arm64/boot/dts/freescale/s32g399a-rdb3.dtsi
@@ -69,8 +69,4 @@ &usdhc0 {
 	mmc-hs400-1_8v;
 	mmc-hs400-enhanced-strobe;
 	disable-wp;
-	/* Remove no-1-8-v to enable higher speed modes for SD card such as: SDR50 or DDR50
-	 * or eMMC memory, such as HS200, HS400 or HS400 ES respectively.
-	 */
-	no-1-8-v;
 };
diff --git a/arch/arm64/boot/dts/freescale/s32gxxxa-evb.dtsi b/arch/arm64/boot/dts/freescale/s32gxxxa-evb.dtsi
index a917f4988222..237a86feb720 100644
--- a/arch/arm64/boot/dts/freescale/s32gxxxa-evb.dtsi
+++ b/arch/arm64/boot/dts/freescale/s32gxxxa-evb.dtsi
@@ -29,7 +29,6 @@ &usdhc0 {
 	mmc-hs200-1_8v;
 	mmc-hs400-1_8v;
 	mmc-hs400-enhanced-strobe;
-	no-1-8-v;
 	status = "okay";
 };
 
-- 
2.45.2

